### ISSUE TITLE
Making language selection easier for non-English speakers

### ISSUE THREAD

**Issue by mairin** [fedoradesign/anaconda#2]
Maybe I'm overly conscious about the plight of someone who can't understand English, but... 

BEFORE
![lang-1](https://cloud.githubusercontent.com/assets/799683/6252631/b8fe4542-b769-11e4-92cd-88ab3ddf5979.png)

AFTER
![lang-2](https://cloud.githubusercontent.com/assets/799683/6252635/bc1a4e06-b769-11e4-9acb-88897b1c8fd2.png)

I'm worried about the "Additional languages" item too - is there a good graphical way to represent that?

**Comment by mairin** [comment]
Some brain food ideas for additional lang selection:

Add icon to option - globe?
![lang-3](https://cloud.githubusercontent.com/assets/799683/6252939/cca2d94e-b76b-11e4-952a-e069fddc7dff.png)

Make expander a button with arrow
![lang-4](https://cloud.githubusercontent.com/assets/799683/6252971/f6e42d0c-b76b-11e4-80b9-6ee969cb7126.png)

**Comment by mairin** [comment]
Okay just some food for thought, because I wanted to get a better idea for how important this is. I translated the first screen (likely very, very badly) to Japanese. The first one below has all text written in Japanese. The second has the languages in their native names whereas the rest of the UI is in Japanese. For me it makes a big difference. I would definitely need to take some extra time trying to figure out how to display more languages if English wasn't listed in the native language in the shortlist - and if I didn't understand enoguh Japanese to know the Kanji for eigo (I took 2 years in college :) ) I think it would be total fail for me to select English here.

![japanese](https://cloud.githubusercontent.com/assets/799683/6253872/94266296-b772-11e4-826c-ecad3bbf00be.png)

![japanese2](https://cloud.githubusercontent.com/assets/799683/6253881/a5107de4-b772-11e4-8be9-d83cdbe22d13.png)

**Comment by mairin** [comment]
And.... in this context, I feel like down arrow button makes more sense than the globe....

![globe](https://cloud.githubusercontent.com/assets/799683/6254275/71b81ddc-b775-11e4-8c70-620162cdbc23.png)

---

![arrow](https://cloud.githubusercontent.com/assets/799683/6254276/752a9f08-b775-11e4-8e59-6eb27a20d54c.png)

**Comment by hbons** [comment]
what about a speech bubble with "..."?
maybe the full scrollable list should just be shown underneath the separator?

**Comment by mairin** [comment]
Here's what it looks like with the speech bubble + ... - I really like that option as well! I think it reads the best to me given a screen full of stuff I don't really understand well.

![japanese5](https://cloud.githubusercontent.com/assets/799683/6256684/1aff3a08-b788-11e4-9ac2-fa1f6489d8cc.png)

Here's some thinking behind having a short list and a longer list vs just a longer list -
- today in production anaconda we show the full list of langs (70+ supported) + a second list to the right for the locales that apply to the selected lang. since the default language is (usually) english you've got these two huge scrollbars by default and it's overwhelming for everybody
- we do have some smarts in there that selects the default lang / bubbles likely langs to the top based on geolocation already. but still, even if we can sort it down to likely candidates to you, you've got two huge lists
- if we can catch say just 50% of users with the geolocation suggestions 'shortlist', then they don't even have to deal with the huge long list
- the huge long list is helpful though if your language isn't in the default set or in the 'shortlist' given if the languages are written in their native name - although its a huge list, your lang's name written natively in that lang likely will jump out at you even in a big long list. but 100% of people dealing with the big long list is maybe a bit much?

**Comment by garrett** [comment]
I do like the look of this overall:

![](https://cloud.githubusercontent.com/assets/799683/6253881/a5107de4-b772-11e4-8be9-d83cdbe22d13.png)

I'm not sure if the speech icon looks clickable enough. The down arrow or item in the list both look more clickable. What about a hybrid where the speech bubble icon is in the list, where the globe is? Here's a mockup of this idea:
![lang-list-bubble](https://cloud.githubusercontent.com/assets/10246/6440937/bbfeb10a-c0e2-11e4-89e1-48d1325d53cc.png)

But, after seeing it, I do like the item as a button. I think it only works embedded in the list like that as a globe (at least so far).

**Comment by garrett** [comment]
(But, yeah, after the quick hybrid above, I do agree with you both that the button approach would work better. I just wanted to see the hybrid mockup, and thought it'd be useful to share.)

**Comment by garrett** [comment]
How many languages are included in a typical region? Are any languages _always_ or almost always on that list? (For example: English, as we know it's a good translation, and there are many people who know English as a second language.)

If there is an extended list of languages that are in addition to the geo-based languages, does it take the common UN languages into account? https://en.wikipedia.org/wiki/Official_languages_of_the_United_Nations (English, Mandarin, Spanish, French, Russian, Arabic)? or demographics https://en.wikipedia.org/wiki/Linguistic_demography? or perhaps even by GDP http://www.unicode.org/notes/tn13/ (which, sadly, is probably indicative of those who have access to computers)?

In other words, someone might know enough English, Spanish, or Mandarin as a second language even if the default list of geo-selected languages is not ideal for them, in order to allow them to more easily change to their preferred language or to install — if the list has more than _only_ their most likely local languages. (Think of the case where someone is installing while on a VPN, for example. The GeoIP based selection would probably omit their country's best selection.)

**Comment by garrett** [comment]
TL;DR: I like the button best too. 

The arrow looks logical to me, and the speech one could also possibly work. The difference, to me, is that the arrow would expand the list in-line, whereas the speech one would probably have a pop-up / lightbox instead. (This is something that should probably be tested, to figure out which of the behaviors and icons options are best.)
